### DISCUSSION TITLE
简悦标注自动同步到 Obsidan 并通过 Excalidraw 从而实现思维导图成体系化的全自动化流程

### DISCUSSION THREAD

**Discussion by windily-cloud** [Kenshin/simpread#3442] [服务]
# 实现效果

1. 在简悦的阅读模式粗读，做好重 点笔记
2. 自动同步笔记到 Obsidian 里
3. Obsidian 通过quickadd脚本，将笔记放到自由度更大的 Excalidraw 里
4. 通过回链精读原文，拖动放置图形元素，组织各元素形成体系化的知识
5. 总结图谱形成永久笔记

# 视频展示

> 从批注到图谱的流程

https://user-images.githubusercontent.com/30640638/154836689-a2fa4ec5-aa56-4cb9-a55f-93bc9eef911d.mov

# Excalidraw 的介绍

![2](https://user-images.githubusercontent.com/30640638/154835843-495a2eb6-9920-4a02-a321-2310f3e6ecf8.jpeg)

图谱非常强大：
1. 支持手写，双链，超链接，图片，latex，md格式。
2. 自定义每个元素的样式，图标自定义。
3. 脚本（结构化文本自动生成思维导图，图片OCR，一键添加连线、图形、图片 、md文档等操作）。
4. 内置手写图形分享社区，方便插入表情，复用已有的元素。
5. 内置脚本社区方便白嫖。
6. 本地存储为md文档，离线可用，随时修改。
7. 一键将图谱生成png或svg格式，方便插入任意md文档。
8. 完全免费，有能力的建议在插件设置里支持下这位大鼻子大叔，这样就能一直用更加强大的插件，哈哈😆😆😆。

# 配置过程

需求：
1. 简悦及简悦高级会员
2. quickadd 插件
3. Excalidraw 插件

### 配置同步助手

- [同步助手配置教程](https://kb.simpread.pro/#/page/%E9%85%8D%E7%BD%AE%E5%90%8C%E6%AD%A5%E5%8A%A9%E6%89%8B)

先下载同步助手：这里简悦拓展版本和同步助手版本须看清重要提示。
![5](https://user-images.githubusercontent.com/30640638/154835870-6ec9eb2b-5ef2-43f1-9d8d-0ac5a506473d.png)

验证同步助手：
1. 运行 **简悦 · 同步助手**
2. **简悦 → 选项页 → 共通 → 简悦 · 同步助手 → 授权验证并开启简悦 · 同步助手**

[增强导出配置教程](https://kb.simpread.pro/#/page/%E5%A2%9E%E5%BC%BA%E5%AF%BC%E5%87%BA)：截个我自己的配置作为参照，具体配置过程还得看文档

![6](https://user-images.githubusercontent.com/30640638/154835874-3f03b94c-6b88-4eac-822a-b73ff27f04a1.png)

![7](https://user-images.githubusercontent.com/30640638/154835876-02a37aa1-020a-446a-8516-cf674a029e57.png)

### 配置定制导出服务

在浏览器拓展->偏好设置->服务里：

![8](https://user-images.githubusercontent.com/30640638/154835880-22b5e204-53ec-40ca-9f2f-aaf1738d544a.png)

修改模板为：你也可以根据[模板配置文档](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown)自己定制

```
# 标注信息
- title：{{title}}
- create：{{create}}
- link：[回链地址]({{url}})

# 标注内容
{{#each}}
- [{{an_text}}]({{an_org_uri}})：{{an_note}}
{{/each}}
```

### 配置自动化

在和定制导出服务下面，配置自动化：
![9](https://user-images.githubusercontent.com/30640638/154835885-9ab3c99a-f868-4db2-bda0-40e4bee9881a.png)

按照图示配置即可。

### 验证是否配置成功

1. 找到一个网页
2. 打开阅读模式
3. 标注：这个时候右上角就会显示保存成功，并在你配置的目录生成markdown文档

## 配置 Obsidian 的 quickadd 插件

- 确保安装了 quickadd 插件
- 创建一个目录用于放置js脚本文件
![10](https://user-images.githubusercontent.com/30640638/154835910-d829e3d4-0bb1-4b9b-9e39-2eee97056189.png)

- 创建一个 JS 文件，命名为 `excalidraw`，当然可以随意命名，这样方便我后续叙述：须在系统文件管理器中创建 JS 文件，内容如下：

```
module.exports = async (params) => {
  const file = app.workspace.activeLeaf.view.file
  const content = await app.vault.cachedRead(file)
  let lines = await content.split("\n").filter(line => line.includes("- ["))

  const ea = ExcalidrawAutomate;
  ea.reset();
  lines.forEach((element,index) => {
    ea.addText(-100,70*index,element.replace(/^-\s/, ""))  
  });
  await ea.create();
}
```

- 在 quickadd 的设置里，点击：
![11](https://user-images.githubusercontent.com/30640638/154835914-4d97c902-31a3-4792-9571-55329c5a6a69.png)

- 创建一个名为`excalidraw-mindmap`的Macro
![12](https://user-images.githubusercontent.com/30640638/154835919-6db91c4b-f44d-42ae-9075-0fef8743b5b0.png)

- 把js文件添加到macro中：
![13](https://user-images.githubusercontent.com/30640638/154835924-3c0db1a3-2285-4e5f-b140-21f3bc04e596.png)

- 回到配置页，进一步设置：
![14](https://user-images.githubusercontent.com/30640638/154835927-b79167bf-af95-4be4-a7c3-25d429ba1d55.png)

- 将macro配置好：
![15](https://user-images.githubusercontent.com/30640638/154835936-f7fdaa4a-4144-4b20-b8a5-26e7fff82421.png)

- 点击闪电按钮将该命令加入到命令面板中：
![16](https://user-images.githubusercontent.com/30640638/154835939-62ba9be9-bf2a-403b-8e5a-e1b05c9d4b98.png)

配置后需要重启 Obsidain

- 在简悦生成的 `md` 文档中，按 `ctrl + p`，选择 quickadd 中 `excalidraw-mindmap` 命令。

- 你将会看到生成的 excalidraw 。

# 总结

Excalidraw 提供高级的选项，包括 OCR 识别，自动化脚本等，理论上结构化的md文档能自动化生成图形，连线，文本，链接等，可折腾性max。简悦，obsidian，quickadd，excalidraw每一个都支持脚本功能，每一个都能让人头秃，衷心感谢这些无私奉献的人，提供如此强大的软件功能！

**Comment by Kenshin** [comment]
哈哈，非常感谢，很有意思的工作流，尤其是引入了 Excalidraw 方案，给了我很多思路。

**Comment by zijie98** [comment]
试了，总是提示Template Path错误

**Reply by windily-cloud** [reply]
能截张图吗？你进行到哪一步我都不知道。

**Comment by Feng-YiJing-Dao** [comment]
在md运行该命令后,生成的excalidraw是空白的...

**Comment by Kenshin** [comment]
嗨，各位。

简悦官方针对新用户或不喜欢配置的用户，发布了一个新功能：**Obsidan 配置库**。

细节请看下面，几分钟就能配置成完成。

***

![img](https://imgs.zhubai.love/c626678ff6b64dd1ac9dfdae6cf6d188.png)

> 针对新用户的一步到位的 Obsdian 配置库。

## 🗂 配置库

[简悦 · 配置库](https://www.yuque.com/kenshin/simpread/ds8zk0) 是简悦官方推出的一套针对新用户的极简配置方案，方便新用户用最快的方式使用简悦的各种高级服务，配置库内置了常用的双链笔记用法，如：Notion、Obsidian、Logseq、Roam Research，同时包含了简悦在阅读模式上的一些常规插件：Live Editor、题图、Safari 阅读模式等。

## 💡 特点

用户仅需要下载对应的配置库，修改里面的 UID 为自己的高级账户 UID，即可实现简悦的各种高级操作。

## 💁‍♂️ 受众

新用户或者不想使用手动配置的用户。

## 📝 如何使用

1️⃣ 下载

2️⃣ 修改为自己的 UID

3️⃣ 将必要的文件夹复制粘贴到相应位置

✅ Done！

## 📚 Obsidian 配置库

## 📝 特点

1️⃣ 加入稍后读，自动将本地快照和 Markdown 导入到 Obsidian

2️⃣ 加入标注时，自动将标注内容导入本地快照和 Obsidian

3️⃣ 改变稍后读或标注的元数据时，自动将改动的内容保存到本地快照和 Obsidian

4️⃣ 在 Obsidian 中直接内置好了简悦的标注系统，可直接在 Obsidian 实现标注

## 📖 区别

![img](https://imgs.zhubai.love/f4a51d8e8fb4401281b07f82bf0bc7d1.png)

## 📘 极速版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@little/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/xkuecp)

不需要同步助手，无任何配置，支持 Markdown 形式的快照，自动同步标注，适合 Obsidain + 简悦的尝鲜用户。

## 📙 基础版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@localrestapi/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/cg33gh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，适合 Obsidain + 简悦的轻量级用户。

## 📗 高级版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@sync/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/wq35mh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，可在 Obsidian 中标注，适合 Obsidain + 简悦的重度用户。

**Comment by brmysss** [comment]
<img width="295" alt="image" src="https://github.com/Kenshin/simpread/assets/41810235/d8bbb989-2d63-4e9a-9695-3bdbf17bb83a">
提示错误
